### PR TITLE
backup: refactor and add test for writeSnap()

### PR DESCRIPTION
**Pull request by hongchaodeng**

### CODE REVIEW

File: pkg/backup/backup_manager.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		namespace:     namespace,
+		etcdTLSConfig: etcdTLSConfig,
+		be:            be,
+	}
+}
+
+// SaveSnap saves the latest snapshot if its revision is greater than the given lastSnapRev
+// and returns a BackupStatus containing saving backup metadata if SaveSnap succeeds.
+func (bm *BackupManager) SaveSnap(lastSnapRev int64) (*backupapi.BackupStatus, error) {
+	podList, err := bm.kclient.Core().Pods(bm.namespace).List(k8sutil.ClusterListOpt(bm.clusterName))
```
Comment (fanminshi) [inline pkg/backup/backup_manager.go]: how do i fake bm.kclient?

File: pkg/backup/backup_manager.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	bs, err := bm.writeSnap(etcdcli.Maintenance, member.ClientURL(), rev)
+	if err != nil {
+		return nil, fmt.Errorf("write snapshot failed: %v", err)
+	}
+	logrus.Infof("saved backup (rev: %v, etcdVersion: %v) for cluster (%s)",
+		bs.Revision, bs.Version, bm.clusterName)
+	return bs, nil
+}
+
+func (bm *BackupManager) writeSnap(mcli clientv3.Maintenance, endpoint string, rev int64) (*backupapi.BackupStatus, error) {
```
Comment (fanminshi) [inline pkg/backup/backup_manager.go]: we are only test `writeSnap` function right?

File: pkg/backup/backup_manager.go (hunk 1 above)
Comment (fanminshi) [reply pkg/backup/backup_manager.go]: @hongchaodeng you are right. I was thinking about testing SaveSnap.

**Comment by fanminshi** [comment]
@hongchaodeng implemented test. ptal.

**Comment by fanminshi** [comment]
All fixed PTAL. cc/ @hongchaodeng

**Comment by hongchaodeng** [comment]
test failed

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**